mipengine/controller/node_landscape_aggregator.py

```diff
@@ -155,97 +129,129 @@ async def update(self):
-                    compatible_data_models, aggregated_datasets
-                )
-                datasets_locations = _get_dataset_locations_of_compatible_data_models(
-                    compatible_data_models, dataset_locations
-                )
+                nodes_info = _get_nodes_info(nodes_addresses)
+
+                with self._update_lock:
+                    self._node_registry.nodes = {
+                        node_info.id: node_info for node_info in nodes_info
+                    }
+
+                    local_nodes = [
+                        node for node in nodes_info if node.role == NodeRole.LOCALNODE
+                    ]
+                    (
+                        dataset_locations,
+                        aggregated_datasets,
+                    ) = _gather_all_dataset_info(local_nodes)
+                    data_model_cdes_per_node = _get_cdes_across_nodes(local_nodes)
+                    compatible_data_models = _get_compatible_data_models(
+                        data_model_cdes_per_node
+                    )
+                    _update_data_models_with_aggregated_datasets(
+                        compatible_data_models, aggregated_datasets
+                    )
+                    datasets_locations = (
+                        _get_dataset_locations_of_compatible_data_models(
+                            compatible_data_models, dataset_locations
+                        )
+                    )
 
-                self._node_registry.nodes = {
-                    node_info.id: node_info for node_info in nodes_info
-                }
+                    self._data_model_registry.data_models = compatible_data_models
+                    self._data_model_registry.datasets_location = datasets_locations
+
+                    logger.debug(
+                        f"Nodes:{[node for node in self._node_registry.nodes]}"
+                    )
+                    print(f"Nodes:{[node for node in self._node_registry.nodes]}")
 
-                self._data_model_registry.data_models = compatible_data_models
-                self._data_model_registry.datasets_location = datasets_locations
-                self._logger.debug(
-                    f"Nodes:{[node for node in self._node_registry.nodes]}"
-                )
             except Exception as exc:
-                self._logger.error(
-                    f"Node Landscape Aggregator exception: {type(exc)}:{exc}"
+                logger.warning(
+                    f"NodeLandscapeAggregator caught an exception but will continue to "
+                    f"update {exc=}"
                 )
+                pass
             finally:
-                await asyncio.sleep(NODE_LANDSCAPE_AGGREGATOR_UPDATE_INTERVAL)
+                time.sleep(NODE_LANDSCAPE_AGGREGATOR_UPDATE_INTERVAL)
+
+    def _initialize(self):
+        self._node_registry = NodeRegistry(logger)
+        self._data_model_registry = DataModelRegistry(logger)
```

Comment:
I guess that the point of this `initialize` is to reset the registries when the `start` method is called right?

Why do we need to `reset` the registries if the NLA is stopped and then started?

Reply:
I have not really checked the internals of `NodeRegistry` and `DataModelRegistry` to make sure that it is not needed to instantiate new objects everytime the `NodeLandscapeAggregator` gets initialized, but these objects are very cheap to instantiate and the initialization of `NodeLandscapeAggregator` will not happen very often, so it seems the safest choice.